packages/build-utils/src/executors/old-builder/old-executor.ts

```diff
@@ -82,6 +95,22 @@ export default async function runExecutor(options: OldBuilderExecutorSchemaType,
   await copyFiles(currentProjectRoot, filesDest, filesList);
   await Promise.all([copyFolder(distSrc, distDest), copyFolder(docsSrc, docsDest)]);
   await removePackage(distDest);
+
+  // NX does not officially expose or document some utility classes such as FsTree which would make this easier
+  // See https://github.com/nrwl/nx/issues/16691
+  const tree: Tree = {
+    root: filesDest,
+    read: readFileSync,
+    write: writeFileSync,
+    exists: existsSync,
+    delete: (path) => unlink(path, () => {}),
+    rename: renameSync,
+    changePermissions: chmodSync,
+    isFile: (path) => lstatSync(path).isFile(),
+    children: readdirSync,
+    listChanges: () => [],
+  };
+  addDependenciesToPackageJson(tree, sharedGlobalDependencies, sharedGlobalDevDependencies, `${filesDest}/package.json`);
```

Comment:
We can avoid the oddness above if we update the package.json manually in lieu of using `addDependenciesToPackageJson` - however there is some logic in the function that will check to see if the dependency needs to be updated/merged (already present in the file).